pkg/actions/lua/databricks/client.go

```diff
@@ -141,24 +148,39 @@ func (client *Client) RegisterExternalTable(l *lua.State) int {
 	if metadata != nil {
 		metadataMap = metadata.(map[string]any)
 	}
-	status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
-	if err != nil {
-		if alreadyExists(err) {
-			err = client.deleteTable(catalogName, schemaName, tableName)
-			if err != nil {
-				lua.Errorf(l, "%s", err.Error())
-				panic("unreachable")
-			}
-			status, err = client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
-			if err != nil {
-				lua.Errorf(l, "%s", err.Error())
-				panic("unreachable")
-			}
-		} else {
+
+	// delete the table first if exists, to prevent "table already exists" errors in databricks
+	if client.checkTableExists(catalogName, schemaName, tableName) {
+		err := client.deleteTable(catalogName, schemaName, tableName)
+		if err != nil {
 			lua.Errorf(l, "%s", err.Error())
 			panic("unreachable")
 		}
 	}
+
+	bo := backoff.NewExponentialBackOff()
+	bo.InitialInterval = 100 * time.Millisecond
+	bo.MaxInterval = 3 * time.Second
+	bo.MaxElapsedTime = 10 * time.Second
+
+	createTableBO := func() (string, error) {
+		status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
+		if err == nil {
+			return status, nil
+		}
+		if alreadyExists(err) {
+			return "", err
+		}
+		return "", backoff.Permanent(err)
+	}
```

Comment:
prefer to check error first
```suggestion
		status, err := client.createExternalTable(warehouseID, catalogName, schemaName, tableName, location, metadataMap)
		if err != nil {
  			if alreadyExists(err) {
  				return "", err
  			}
  			return "", backoff.Permanent(err)
		}
		return status, nil
	}
```